frontend/src/component/project/ProjectList/ProjectList.tsx

```diff
@@ -131,32 +131,34 @@ export const ProjectListNew = () => {
     const { projects, loading, error, refetch } = useProjects();
     const [fetchedProjects, setFetchedProjects] = useState<projectMap>({});
     const { isOss } = useUiConfig();
-
     const isSmallScreen = useMediaQuery(theme.breakpoints.down('md'));
-    const [searchParams, setSearchParams] = useSearchParams();
-    const [searchValue, setSearchValue] = useState(
-        searchParams.get('search') || '',
-    );
-
     const showProjectFilterButtons = useUiFlag('projectListFilterMyProjects');
-    const filters = ['All projects', 'My projects'];
-    const [filter, setFilter] = useState(filters[0]);
     const myProjects = new Set(useProfile().profile?.projects || []);
 
-    useEffect(() => {
-        const tableState: PageQueryType = {};
-        if (searchValue) {
-            tableState.search = searchValue;
-        }
+    const filterOptions = [
+        {
+            label: 'All projects',
+            value: false,
+        },
+        {
+            label: 'My projects',
+            value: true,
+        },
+    ];
 
-        setSearchParams(tableState, {
-            replace: true,
-        });
-    }, [searchValue, setSearchParams]);
+    const stateConfig = {
+        myProjects: withDefault(BooleanParam, false),
```

Comment:
I'm not convinced this should be a boolean. Is there a reason you didn't use an enum? E.g. `projectFilter: 'all projects' | 'my projects'` or `projectFilter: undefined | 'my projects'`?

I'm thinking about this primarily from two perspectives:

1. From a future extension point of view. If we decide we want to add more filters, then tacking on another enum value is easy. But if we use a boolean we need a new property and then need to consider how `myProjects` interacts with the new property.

2. For code clarity: it was not clear to me why `filterOptions` had values `true` and `false`. I at first assumed it had to do with the default value or something.